lib/wasix/src/runners/wasi_common.rs

```diff
@@ -85,13 +87,15 @@ impl CommonWasiOptions {
     }
 }
 
+type ContainerFs =
+    OverlayFileSystem<TmpFileSystem, [RelativeOrAbsolutePathHack<Arc<dyn FileSystem>>; 1]>;
+
 fn prepare_filesystem(
+    root_fs: TmpFileSystem,
     mapped_dirs: &[MappedDirectory],
     container_fs: Arc<dyn FileSystem>,
     builder: &mut WasiEnvBuilder,
-) -> Result<Box<dyn FileSystem + Send + Sync>, Error> {
-    let root_fs = RootFileSystemBuilder::default().build();
-
+) -> Result<ContainerFs, Error> {
```

Comment:
It might be better to return a `impl FileSystem + Send + Sync` here. Something I do quite frequently when debugging is to add a `let fs = virtual_fs::TraceFileSystem::new(fs);` at the bottom of the function so all filesystem calls get logged, and updating the return type every time would be annoying.